sounddevice.py

```diff
@@ -2349,6 +2399,157 @@ def __init__(self, channel_selectors):
             channelSelectors=self._selectors))
 
 
+class MacCoreSettings(object):
+
+    def __init__(self, channel_selectors=None, change_dev_params=False, allow_rate_conv=True, rate_conv_quality='Max'):
+        """Mac Core Audio-specific input/output settings.
+
+        Objects of this class can be used as *extra_settings* argument
+        to `Stream()` (and variants) or as `default.extra_settings`.
+
+        Parameters
+        ----------
+        channel_selectors : iterable of int, optional
+            Support for opening only specific channels of a MacCore
+            device.  *channel_selectors* is a list of integers
+            specifying the (zero-based) channel numbers to use.  Core
+            Audio refers to this as a Channel Map, and this class
+            provides some minor translation to be API compatible with
+            AsioSettings.
+        change_dev_params : bool, optional
+            If ``True``, allows PortAudio to change things like the
+            device's frame size, which allows for much lower latency,
+            but might disrupt the device if other programs are using it,
+            even when you are just Querying the device.  ``False`` is
+            the default.
+        allow_rate_conv : bool, optional
+            In combination with the above flag, ``True`` causes the
+            stream opening to fail, unless the exact sample rates are
+            supported by the device.
+        rate_conv_quality : string, optional
+            This sets MacCore's sample rate conversion quality.
+            Possible values are: ``'Min'``, ``'Low'``, ``'Medium'``,
+            ``'High'``, or ``'Max'``.  ``'Max'`` is the default.
+
+        Examples
+        --------
+        Setting output channels when calling `play()`: (Output to 'Phones 9' and 'Phones 10' on RME Fireface UFX+)
+
+        >>> import sounddevice as sd
+        >>> maccore_out = sd.MacCoreSettings(channel_selectors=[8, 9])
+        >>> sd.play(..., extra_settings=maccore_out)
+
+        Setting default output channels:
+
+        >>> sd.default.extra_settings = maccore_out
+        >>> sd.play(...)
+
+        Setting input channels as well:  (Input from 'Mic 9' and 'Mic 10' on RME Fireface UFX+)
+
+        >>> maccore_in = sd.MacCoreSettings(channel_selectors=[8, 9])
+        >>> sd.default.extra_settings = maccore_in, maccore_out
+        >>> sd.playrec(..., channels=1, ...)
```

Comment:
I don't understand this example. You set channels_selectors=[8,9] but then specify channels=1 to playrec? Wouldn't you want to pass channels=2?